mqt/qecc/cc_decoder/color_code.py

```diff
@@ -0,0 +1,61 @@
+"""A color code class."""
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+import numpy as np
+from ldpc import mod2
+
+if TYPE_CHECKING:  # pragma: no cover
+    import numpy.typing as npt
+
+
+class LatticeType(Enum):
+    """The type of lattice used for the color code."""
+
+    HEXAGON = "hexagon"
+    SQUARE_OCTAGON = "square_octagon"
+
+
+class ColorCode:
+    """A base class for color codes on a three-valent, three-colourable lattice."""
+
+    def __init__(self, distance: int, lattice: LatticeType) -> None:
+        """Initialize the color code."""
+        self.distance = distance
+        self.ancilla_qubits: set[tuple[int, int]] = set()
+        self.data_qubits: set[tuple[int, int]] = set()
+        self.qubits_to_faces: dict[int, list[int]] = {}
+        self.faces_to_qubits: dict[int, list[int]] = {}
+        self.lattice = lattice
+        self.add_qubits()
+        self.H = np.zeros((len(self.ancilla_qubits), len(self.data_qubits)), dtype=int)
+        self.construct_layout()
+        self.compute_logical()
+        self.n = len(self.qubits_to_faces)
+        self.k = self.L.shape[1]
```

Comment:
```suggestion
    def __init__(self, distance: int, lattice_type: LatticeType) -> None:
        """Initialize the color code."""
        self.distance = distance
        self.ancilla_qubits: set[tuple[int, int]] = set()
        self.data_qubits: set[tuple[int, int]] = set()
        self.qubits_to_faces: dict[int, list[int]] = {}
        self.faces_to_qubits: dict[int, list[int]] = {}
        self.lattice_type = lattice_type
        self.add_qubits()
        self.H = np.zeros((len(self.ancilla_qubits), len(self.data_qubits)), dtype=int)
        self.construct_layout()
        self.compute_logical()
        self.n = len(self.qubits_to_faces)
        self.k = self.L.shape[1]
```
Note that this will require further changes throughout.